controllers/openstackcontrolplane_controller.go

```diff
@@ -364,6 +367,9 @@ func (r *OpenStackControlPlaneReconciler) Reconcile(ctx context.Context, req ctr
 		osc.Spec.GitSecret = instance.Spec.GitSecret
 		osc.Spec.RunUID = openstackclient.CloudAdminUID
 		osc.Spec.RunGID = openstackclient.CloudAdminGID
+		if instance.Spec.DomainName != "" {
+			osc.Spec.DomainName = instance.Spec.DomainName
+		}
```

Comment:
TODO: how to pass the domainname through to the BMset

Reply:
Might just need to use the new `common.GetControlPlane` func within the BaremetalSet controller to get access to the associated ControlPlane resource